scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/EcdhPskServerKeyExchange.java

```diff
@@ -0,0 +1,247 @@
+		this(curveId, peerAddress);
+		if (pointEncoded == null) {
+			throw new NullPointerException("ephemeral public key cannot be null");
+		}
+		this.pointEncoded = Arrays.copyOf(pointEncoded, pointEncoded.length);
+		// re-create public key from params
+		SupportedGroup group = SupportedGroup.fromId(curveId);
+		if (group == null || !group.isUsable()) {
+			throw new HandshakeException(
+				String.format("Server used unsupported elliptic curve (%d) for ECDH", curveId),
+				new AlertMessage(AlertLevel.FATAL, AlertDescription.HANDSHAKE_FAILURE, peerAddress));
+		} else {
+			try {
+				point = ECDHECryptography.decodePoint(pointEncoded, group.getEcParams().getCurve());
+				KeyFactory keyFactory = KeyFactory.getInstance(KEYPAIR_GENERATOR_INSTANCE);
+				publicKey = (ECPublicKey) keyFactory.generatePublic(new ECPublicKeySpec(point, group.getEcParams()));
+			} catch (GeneralSecurityException e) {
+				LOGGER.debug("Cannot re-create server's public key from params", e);
+				throw new HandshakeException(
+					String.format("Cannot re-create server's public key from params: %s", e.getMessage()),
+					new AlertMessage(AlertLevel.FATAL, AlertDescription.INTERNAL_ERROR, peerAddress));
+			}
+		}
+	}
+
+	private EcdhPskServerKeyExchange(int namedCurveId, InetSocketAddress peerAddress) {
+		super(peerAddress);
+		this.curveId = namedCurveId;
+	}
+
+	// TODO this is called 4 times for Flight 4
+	@Override
+	public byte[] fragmentToByteArray() {
+		DatagramWriter writer = new DatagramWriter();
+		switch (curveType) {
+		// TODO add support for other curve types
+		case EXPLICIT_PRIME:
+		case EXPLICIT_CHAR2:
+			break;
+		case NAMED_CURVE:
+			writeNamedCurve(writer);
+			break;
+		default:
+			LOGGER.warn(MSG_UNKNOWN_CURVE_TYPE, curveType);
+			break;
+		}
+		return writer.toByteArray();
+	}
+	
+	/** ServerKeyExchange message is always sent even if psk identity hint is not present.<br>
+	 * See <a href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a> */
+	private void writeNamedCurve(DatagramWriter writer) {
+		writer.write(NAMED_CURVE, CURVE_TYPE_BITS);
+		writer.write(curveId, NAMED_CURVE_BITS);
+		writer.write(pointEncoded.length, PUBLIC_LENGTH_BITS);
+		writer.writeBytes(pointEncoded);
+	}
+	
+	/**
+	 * Deserialize byte array to key exchange message.
+	 * 
+	 * @param byteArray
+	 * @param peerAddress
+	 * @return {@code EcdhPskServerKeyExchange}
+	 * @throws HandshakeException - if the byte array includes unsupported curve
+	 * @throws NullPointerException if byteArray is {@code null}
+	 */
+	public static HandshakeMessage fromByteArray(byte[] byteArray, InetSocketAddress peerAddress) throws HandshakeException {
+		if (byteArray == null) {
+			throw new NullPointerException("byte array cannot be null");
+		}
+		DatagramReader reader = new DatagramReader(byteArray);
+		int curveType = reader.read(CURVE_TYPE_BITS);
```

Comment:
IMHO this will fail because the byte array will contain two bytes indicating the identity hint length at this position ...